crates/wash-cli/src/dev.rs

```diff
@@ -152,25 +319,482 @@ impl KnownDep {
+
+    /// Configurations that must be created and/or consumed by this dependency
+    configs: Vec<wadm_types::ConfigProperty>,
+
+    /// Secrets that must be created and/or consumed by this dependency
+    ///
+    /// [`SecretProperty`] here support a special `policy` value which is 'env'.
+    /// Paired with a key that looks like "$SOME_VALUE", the value will be extracted from ENV *prior* and
+    secrets: Vec<wadm_types::SecretProperty>,
+}
+
+impl DependencySpecInner {
+    /// Retrieve the name for this dependency
+    fn name(&self) -> String {
+        match self.image_ref.as_deref() {
+            Some(DEFAULT_KEYVALUE_PROVIDER_IMAGE) => "keyvalue-nats".into(),
+            Some(DEFAULT_HTTP_CLIENT_PROVIDER_IMAGE) => "http-client".into(),
+            Some(DEFAULT_HTTP_SERVER_PROVIDER_IMAGE) => "http-server".into(),
+            Some(DEFAULT_BLOBSTORE_FS_PROVIDER_IMAGE) => "blobstore-fs".into(),
+            Some(DEFAULT_MESSAGING_NATS_PROVIDER_IMAGE) => "messaging-nats".into(),
+            _ => format!("custom-{}-{}", self.wit.0, self.wit.1),
         }
     }
+
+    /// Retrieve the image reference for this dependency
+    fn image_ref(&self) -> Option<&str> {
+        self.image_ref.as_deref()
+    }
+
+    /// Retrieve whether this dependency spec is a component
+    fn is_component(&self) -> bool {
+        self.is_component
+    }
+}
+
+// #[derive(Debug, Clone, PartialEq, Eq)]
+// enum SecretSpec {
+//     /// Secrets that {
+//     NamedSecret(S),
+//     /// Secrets that are specified as strings
+//     /// These should *disallowed* from being used outside of development contexts
+//     String,
+//     /// Secrets specified as environment variables
+//     /// these are resolved at the time of execution of `wash dev`
+//     EnvVar(String),
+// }
+
+/// Information related to the dependencies of a given project
+///
+/// Projects can either be inside workspaces, or not (single component/provider).
+#[derive(Default, Debug, Clone, PartialEq, Eq)]
+struct ProjectDeps {
+    /// ID of a session
+    ///
+    /// This is normally used when generating sessions to be used with `wash dev`, in order
+    /// to make sure various dependencies and related files can be uniquely identified.
+    pub(crate) session_id: Option<String>,
+
+    /// Lookup of dependencies by project key, with lookups into the pool
+    dependencies: HashMap<ProjectDependencyKey, DependencySpec>,
+
+    /// Dependencies that receive invocations for given interfaces (i.e. `keyvalue-nats` receiving a `wasi:keyvalue/get`)
+    ///
+    /// The keys to this structure are indices into the `dependencies` member.
+    receivers: HashMap<(WitNamespace, WitPackage, WitInterface, Option<Version>), Vec<usize>>,
+
+    /// Dependencies that perform invocations for given interfaces (i.e. `http-server` invoking a component's `wasi:http/incoming-handler` export)
+    invokers: HashMap<(WitNamespace, WitPackage, WitInterface, Option<Version>), Vec<usize>>,
```

Comment:
same comment re: importers/exporters, just making sure wherever we land we change this too

Reply:
Done!